moto/ec2/models.py

```diff
@@ -6351,6 +6500,152 @@ def describe_transit_gateway_vpc_attachments(
 
         return transit_gateways_attachments
 
+    def delete_transit_gateway_vpc_attachment(self, transit_gateway_attachment_id=None):
+        self.transit_gateway_attachments[
+            transit_gateway_attachment_id
+        ].state = "deleted"
+        transit_gateway_attachment = self.transit_gateway_attachments.pop(
+            transit_gateway_attachment_id
+        )
+        return transit_gateway_attachment
+
+    def modify_transit_gateway_vpc_attachment(
+        self,
+        add_subnet_ids=None,
+        options=None,
+        remove_subnet_ids=None,
+        transit_gateway_attachment_id=None,
+    ):
+        if remove_subnet_ids:
+            self.transit_gateway_attachments[
```

Comment:
nit: could be simplified/shortened to:
```
        tgw_attachment = self.transit_gateway_attachments[transit_gateway_attachment_id]

        if remove_subnet_ids:
            tgw_attachment.subnet_ids = [
                id
                for id in tgw_attachment.subnet_ids
                if id not in remove_subnet_ids
            ]
```
... and `tgw_attachment` can then also be used further below in lines 6524, 6530, etc